pages/api/upload/index.js

```diff
@@ -0,0 +1,34 @@
+import formidable from "formidable";
+import cloudinary from "cloudinary";
+
+cloudinary.config({
+  cloud_name: process.env.CLOUDINARY_CLOUD_NAME,
+  api_key: process.env.CLOUDINARY_API_KEY,
+  api_secret: process.env.CLOUDINARY_SECRET,
+});
+
+export const config = {
+  api: {
+    bodyParser: false,
+  },
+};
+
+export default async function handler(request, response) {
+  if (request.method !== "POST") {
+    return response.status(400).json({ message: "Method not allowed" });
+  }
+
+  const form = formidable({});
+
+  const [fields, files] = await form.parse(request);
+
+  const file = files.cover[0];
```

Comment:
Maybe we want to give a hint that cover here is the name of the input?
```suggestion
  // we need to access files.cover here since it is the name of our file input. Replace this with the name of your
  input.
  const file = files.cover[0];
```

Reply:
ah yes, very good point. I added it 👌